src/main/java/org/entando/plugins/pda/pam/service/group/KieGroupService.java

```diff
@@ -8,31 +8,31 @@
 import org.entando.plugins.pda.core.service.group.GroupService;
 import org.entando.plugins.pda.pam.service.api.CustomQueryService;
 import org.entando.plugins.pda.pam.service.api.KieApiService;
+import org.entando.plugins.pda.pam.service.process.model.KieProcessId;
 import org.kie.server.api.model.definition.AssociatedEntitiesDefinition;
 import org.kie.server.client.ProcessServicesClient;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class KieGroupService implements GroupService {
-
-    public static final String ID_SEPARATOR = "@";
-    private static final int COMPOSED_ID_SIZE = 2;
-
     private final CustomQueryService customQueryService;
     private final KieApiService kieApiService;
 
     @Override
-    public List<String> list(Connection connection, String processId) {
-        String[] composedId = processId == null ? new String[0] : processId.split(ID_SEPARATOR);
-        if (composedId.length == COMPOSED_ID_SIZE) {
-            ProcessServicesClient processServicesClient = kieApiService.getProcessServicesClient(connection);
-            AssociatedEntitiesDefinition associatedEntityDefinitions = processServicesClient
-                    .getAssociatedEntityDefinitions(composedId[0], composedId[1]);
-            String[] inGroups = associatedEntityDefinitions.getAssociatedEntities().values().stream()
-                    .flatMap(Arrays::stream).toArray(String[]::new);
-            return inGroups.length == 0 ? Collections.emptyList() : customQueryService.getGroups(connection, inGroups);
+    public List<String> list(Connection connection, String id) {
+        if (id == null) {
+            return customQueryService.getGroups(connection);
         }
-        return customQueryService.getGroups(connection);
+
+        KieProcessId compositeId = new KieProcessId(id);
```

Comment:
The processId here is the process definition. It is a string, so it cannot be used with this KieProcessId object.